spark/v3.4/spark/src/test/java/org/apache/iceberg/spark/source/TestMetadataTableReadableMetrics.java

```diff
@@ -190,6 +191,7 @@ private GenericRecord createNestedRecord(Long longCol, Double doubleCol) {
   }
 
   @Test
+  @Ignore
```

Comment:
Have disabled this UT purposefully as it failed in one of my CI runs  with error below : 

```


TestMetadataTableReadableMetrics > testPrimitiveColumns FAILED
    java.lang.AssertionError: Row should match: row 1 col 1 (nested col 1) (nested col 2) contents should match expected:<44> but was:<43>
        at org.junit.Assert.fail(Assert.java:89)
        at org.junit.Assert.failNotEquals(Assert.java:835)
        at org.junit.Assert.assertEquals(Assert.java:120)
        at org.apache.iceberg.spark.SparkTestHelperBase.assertEquals(SparkTestHelperBase.java:84)
        at org.apache.iceberg.spark.SparkTestHelperBase.assertEquals(SparkTestHelperBase.java:81)
        at org.apache.iceberg.spark.SparkTestHelperBase.assertEquals(SparkTestHelperBase.java:81)
        at org.apache.iceberg.spark.SparkTestHelperBase.assertEquals(SparkTestHelperBase.java:66)
        at org.apache.iceberg.spark.source.TestMetadataTableReadableMetrics.testPrimitiveColumns(TestMetadataTableReadableMetrics.java:232)
```

will take a deeper look, shortly as the failure seems unrelated

Reply:
This is actually interesting and happening only in java 17 env : 
basically the parquet file written has diff `column_size` value in `files` metadata table, in java 8 it's 44 and in java 17 it's 43.
effectively this line here produces diff results: 
https://github.com/apache/iceberg/blob/d04efee702fcdcdbe3659c12f7442f5000aa246a/parquet/src/main/java/org/apache/iceberg/parquet/ParquetUtil.java#L127


Is it because java 17 provides better compression than java 8 ? as per this blog : https://dkomanov.medium.com/java-compression-performance-fb373078cfde

since this is a value we are getting directly from parquet footer, so effectively we are reading what were writing stats are not getting messed up in between, which seems correct to me. But I might be wrong here, will wait for other folks feedback here if there is deeper investigation required.

Reply:
made the changes to directly fetch column size from data file created, to get rid of hard coded sizes